src/tiff.rs

```diff
@@ -46,14 +49,38 @@ impl<R> TiffDecoder<R>
             tiff::ColorType::RGBA(8) => ColorType::Rgba8,
             tiff::ColorType::RGBA(16) => ColorType::Rgba16,
 
-            tiff::ColorType::Palette(n) | tiff::ColorType::Gray(n) =>
-                return Err(ImageError::UnsupportedColor(ExtendedColorType::Unknown(n))),
-            tiff::ColorType::GrayA(n) =>
-                return Err(ImageError::UnsupportedColor(ExtendedColorType::Unknown(n*2))),
-            tiff::ColorType::RGB(n) =>
-                return Err(ImageError::UnsupportedColor(ExtendedColorType::Unknown(n*3))),
-            tiff::ColorType::RGBA(n) | tiff::ColorType::CMYK(n) =>
-                return Err(ImageError::UnsupportedColor(ExtendedColorType::Unknown(n*4))),
+            tiff::ColorType::Palette(n) | tiff::ColorType::Gray(n) => {
+                return Err(ImageError::Unsupported(
+                    UnsupportedError::from_format_and_kind(
+                        ImageFormat::Tiff.into(),
+                        UnsupportedErrorKind::Color(ExtendedColorType::Unknown(n)),
+                    ),
+                ));
+            }
+            tiff::ColorType::GrayA(n) => {
+                return Err(ImageError::Unsupported(
+                    UnsupportedError::from_format_and_kind(
+                        ImageFormat::Tiff.into(),
+                        UnsupportedErrorKind::Color(ExtendedColorType::Unknown(n * 2)),
+                    ),
+                ));
+            }
+            tiff::ColorType::RGB(n) => {
+                return Err(ImageError::Unsupported(
+                    UnsupportedError::from_format_and_kind(
+                        ImageFormat::Tiff.into(),
+                        UnsupportedErrorKind::Color(ExtendedColorType::Unknown(n * 3)),
+                    ),
+                ));
+            }
+            tiff::ColorType::RGBA(n) | tiff::ColorType::CMYK(n) => {
+                return Err(ImageError::Unsupported(
+                    UnsupportedError::from_format_and_kind(
+                        ImageFormat::Tiff.into(),
+                        UnsupportedErrorKind::Color(ExtendedColorType::Unknown(n * 4)),
+                    ),
+                ));
+            }
```

Comment:
Can you define a short local helper function to produce this error?

Reply:
Added [`err_unknown_color_type(u8) -> ImageError`](https://github.com/image-rs/image/compare/249fd6f456cd998e985b2de3f58d9f0dea76f8e8..77795190f4117a4a30677b331afe2767875ed979#diff-2b94ab211a059bc70a3fbd97d84472a3R70-R76) and used it.